linera-views/src/key_value_store_view.rs

```diff
@@ -409,6 +409,11 @@ where
         if self.was_cleared {
             return Ok(None);
         }
+        let iter = self.deleted_prefixes.iter();
+        let mut lower_bound = GreatestLowerBoundIterator::new(0, iter);
+        if !lower_bound.is_index_absent(index) {
+            return Ok(None);
+        }
```

Comment:
That will needlessly iterate over all entries `<= index`. Would this be equivalent?

```suggestion
        if self.deleted_prefixes.range(get_interval(index.clone()).next().is_none() {
            return Ok(None);
        }
```

Reply:
No.
The code that I proposed is looking for a prefix p such that p <= x, that is p is a prefix of x.

The code that you propose is looking for a prefix p such that x <= p, that is x is a prefix of p.

Reply:
Ah, right! :+1: